File: js/src/data/Layer.ts
import { Feature } from './Feature';

export class Layer {
    name: string;
    version: number;
    length: number;
    extent: number;
    features: Feature[];

    constructor(name: string, version : number, extent : number, features: Feature[]) {
        this.name = name;
        this.length = features.length;
        this.features = features;
        this.version = version;
        this.extent = extent;
    }

    public feature = (i) => {
      if (i < 0 || i >= this.length) throw new Error('feature index out of bounds');
      return this.features[i];
    }
}

File: cpp/src/mlt/layer.cpp
#include <mlt/layer.hpp>
#include <mlt/geometry_vector.hpp>

namespace mlt {

Layer::Layer(std::string name_,
             int version_,
             extent_t extent_,
             std::unique_ptr<geometry::GeometryVector>&& geometryVector_,
             std::vector<Feature> features_,
             PropertyVecMap properties_) noexcept
    : name(std::move(name_)),
      version(version_),
      extent(extent_),
      geometryVector(std::move(geometryVector_)),
      features(std::move(features_)),
      properties(std::move(properties_)) {}

Layer::~Layer() = default;

} // namespace mlt
